django/db/models/query_utils.py

```diff
@@ -537,13 +537,41 @@ def relabeled_clone(self, change_map):
 
     def resolve_expression(self, query, reuse, *args, **kwargs):
         clone = self.clone()
-        clone.resolved_condition = query.build_filter(
-            self.condition,
-            can_reuse=reuse,
-            allow_joins=True,
-            split_subq=False,
-            update_join_types=False,
-        )[0]
+
+        if clone.resolved_condition is not None:
+            return clone
+
+        if not hasattr(query, "_resolving_relations"):
+            query._resolving_relations = {}
+
+        relation_key = self.relation_name
+
+        if query._resolving_relations.get(relation_key, False):
+            from django.db.models.sql.where import WhereNode
+
+            clone.resolved_condition = WhereNode()
```

Comment:
Won't that assign an empty `WhereNode` if the relation is referenced in other locations?